### ISSUE TITLE
[1.3.13] FR: Add a validate command

### ISSUE THREAD

**Issue by kbrandwijk** [graphcool/graphcool-cli#235]
I don't know if it's mentioned already somewhere, or if it's similar to doing a 'dry run', but it would be nice to be able to run a validate command that would check:
- Project file syntax
- Permission and function types found in schema
- Graphql queries/PQ/SE SDL valid
- ...?

**Comment by Merott** [comment]
I was going to open a feature request for a DRY-run option, and I came across this.

My use case is that I want to be able to run the dry-run the `deploy` command as part of my CI build process, in order to ensure that the deployment would succeed once a feature branch is merged into master.

At the moment `graphcool diff` is the closest thing we have to this, as it achieves validation, but it doesn't verify that the deployment would succeed. For example, if it's a destructive deployment because of a removed field or type, `deploy` would fail unless `--force` is used, but not `diff`.

I would propose a `--dry-run` option for `graphcool deploy`. Should I open a separate issue for this?

**Comment by kbrandwijk** [comment]
@Merott Thanks for the valuable addition. Especially the destructive deployment part.

**Comment by marktani** [comment]
I think this issue is great to discuss a dry run feature 🙂 

> For example, if it's a destructive deployment because of a removed field or type, deploy would fail unless --force is used, but not diff.

I thought `diff` is already covering this, but that's a great point 👌
